### ISSUE TITLE
Project disappeared (delbyusr unexpectedly) & no version had data to recover it with (using internal tools script)

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1148]
Report from user of students' work disappearing in a **shared** account:

> This past week, some of the projects my students made seem to have disappeared.  Without going into too much detail, I can say with confidence that I don't think it's because somebody deleted any of the projects.  At least one project disappeared very suddenly today while a student logged out and logged back in about 2 seconds later to find their project gone.
> I did a bit of research into the discussion forums, and found a similar problem may have occurred to other users.  One user described saving their work, logging out, and logging back in immediately, and suddenly their project had changed - it was now titled "untitled" and all the sprites and scripts had disappeared, but the project still showed loading multiple assets (which wouldn't normally happen on a blank untitled project).  This sounds exactly like what happened to my student today...

I tried to recover their project, 114360338, but there was only 1 version of the project found by @cwillisf's script. When I uploaded that recovered project as 114581248, it had no data in it (I put it in the user's trash & told them it was there in case they wanted to see it).

Not sure if any relation to https://github.com/LLK/scratch-flash/issues/1110.

**Comment by jwzimmer** [comment]
- Sounds like this user may have seen the same thing: https://scratch.mit.edu/discuss/topic/1008/ ("mysterious blank project")

**Comment by jwzimmer** [comment]
Additional info from reporting user:
> It actually seems that all the projects mysteriously disappeared overnight [not just the one project mentioned above].  It's really odd, because knowing this particular group of students I really don't think one of the students did it.

**Comment by jwzimmer** [comment]
Sense the user reported that all the projects mysteriously disappeared (meaning they clearly didn't want them to be deleted), I restored all 192 projects that had "delbyusr" status.

Some of them are blank (no Scratch Cat even) e.g. 113107533 which could have been caused by a bug (or a person, of course).

Some of the projects have assets but no scripts, e.g. 113214138, which seems a little unusual as an intentional workflow (again, could be a bug causing data loss, or a person).

**Comment by jwzimmer** [comment]
Another user reported all their projects disappearing via freshdesk ticket: 51288.

**Comment by jwzimmer** [comment]
Freshdesk no. 53698: I restored all projects for this user that were delbyusr from 7/22 or earlier.

> I have been frequently losing work (sometimes hours and hours of work) randomly. This has happened three times now and it happens after I log back into my account on a new day.

**Comment by ntlrsk** [comment]
@jwzimmer I wonder if this person has confirmed their email? We had someone from SETC who had issues with one of his projects that went away when he confirmed (not sure if it was a coincidence but thought to mention it).

**Comment by jwzimmer** [comment]
@ntlrsk Thanks, I will ask the user! : )

**Comment by jwzimmer** [comment]
@ntlrsk - I checked in the admin panel, and their email is confirmed. Thanks for bringing this up!

**Comment by jwzimmer** [comment]
Freshdesk no. 53696: I restored all projects for this user that were delbyusr from 7/22 or earlier (all of the more recent projects I could see were trshbyusr or visible already).

I found two projects that seemed to match the two they described in their email as missing when I restored the projects (they were 'delbyusr').

@cwillisf - Since the delbyusr projects for this user & the penultimate user were both from 7/22 or earlier, is it possible something happened around 7/22 that caused the projects to change status? (Assuming some of these projects are the missing ones, which they might not be! I have reached out to the users for verification.)

**Comment by colbygk** [comment]
Based on the freshdesk ticket, I've looked at logentries to see mhirning1's access to scratch.mit.edu if I could figure out which project ID was involved for this report.

Of the handful of projects it appears they accessed, I cannot find one that they own. It would be useful to get info on which project the user was editing.

**Comment by jwzimmer** [comment]
@colbygk I asked them for specific titles/ IDs, I will update the case with that info if they respond. : )

**Comment by joker314** [comment]
Could https://scratch.mit.edu/discuss/topic/216392/ have been caused by this?

**Comment by jwzimmer** [comment]
@joker314 Sounds like it could be, thanks for bringing that thread to our attention. : )

**Comment by jwzimmer** [comment]
I have not heard back from the reporting user in freshdesk no. 53698.

**Comment by jwzimmer** [comment]
A user reported projects disappearing from a **shared** account via freshdesk no. 59726:
> My class made projects and saved them into out scratch account (username = [see freshdesk ticket]). Can you explain what has happened please?

**Comment by jwzimmer** [comment]
Another example of projects getting `delbyusr` status without the user deleting them: https://github.com/LLK/scratchr2/issues/4167.

When I restored two of the projects (the two projects in the db that had `delbyusr` status for one of the users in that case), the teacher (reporting user) said that the student looked at the projects and they were not as she expected:
> That is not what the finished product looked like at all. The script is all jumbled and messed up [see other case for username]. She didn't erase this project because she was excited to show the class and then it was gone from her account. [See other case for username] had a simmilar problem with projects being gone.

**Comment by jwzimmer** [comment]
This user reported their projects being gone (there are 48 deleted projects) and their studio being gone, in freshdesk no. 71422:

> I have been a regular user of Scratch since Sep 2015. I have many projects, 1 studio and was a curator for another studio. 
On Thu, Feb 09, when I tried to login to Scratch, I was repeatedly challenged as a robot. I typed in userid/password, and kept getting the Captcha. Even when I entered the userid/password and Captcha correctly, I could NOT login. 
Finally I reset my password and was able to login to Scratch. 
But, all my projects and studio were gone. 
I never deleted any projects. 
Is there some way to recover my projects?
